libs/books/data-access/src/lib/+state/reading-list.reducer.spec.ts

```diff
@@ -51,6 +51,32 @@ describe('Books Reducer', () => {
 
       expect(result.ids).toEqual(['A', 'B', 'C']);
     });
+
+    it('markBookAsFinished should mark book as finished in state', () => {
+      const finishedDate = new Date().toISOString();
+      const item = createReadingListItem('B');
+      const action = ReadingListActions.markBookAsFinished({
+        bookId: item.bookId,
+        finishedDate
+      });
+
+      const result: State = reducer(state, action);
+
+      expect(result.entities[item.bookId].finishedDate).toEqual(finishedDate);
+      expect(result.entities[item.bookId].finished).toBe(true);
+    });
+
+    it('markBookAsFinishFailed should undo book marked as finished in state', () => {
+      const item = createReadingListItem('B');
+      const action = ReadingListActions.markBookAsFinishFailed({
+        bookId: item.bookId
+      });
+
+      const result: State = reducer(state, action);
+
+      expect(result.entities[item.bookId].finishedDate).toEqual('');
```

Comment:
would this be equal to an empty string, or an undefined value? I am mostly referring to the reducer code, maybe it should set finishedDate as undefined? If we were to also get rid of the `finished` attribute.